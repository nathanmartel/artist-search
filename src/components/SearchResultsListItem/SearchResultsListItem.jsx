import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import './SearchResultsListItem.css';

const SearchResultsListItem = ({ id, name }) => {
  return (
    <li key={id}>
      <Link to={`/artist/${id}`}>{ name }</Link>
    </li>
  );
};

SearchResultsListItem.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired
};

export default SearchResultsListItem;
