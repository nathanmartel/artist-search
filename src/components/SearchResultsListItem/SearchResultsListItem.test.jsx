import React from 'react';
import { shallow } from 'enzyme';
import SearchResultsListItem from './SearchResultsListItem';

describe('SearchResultsListItem component', () => {
  it('renders SearchResultsListItem', () => {
    const wrapper = shallow(<SearchResultsListItem id={'idstring'} name={'Radiohead'} />);
    expect(wrapper).toMatchSnapshot();
  });
});
