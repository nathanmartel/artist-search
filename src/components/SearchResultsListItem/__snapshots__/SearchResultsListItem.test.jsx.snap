// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultsListItem component renders SearchResultsListItem 1`] = `
<li
  key="idstring"
>
  <Link
    to="./Radiohead"
  >
    Radiohead
  </Link>
</li>
`;
